### PR TITLE
Add MSQ and SIWS pages

### PR DESCRIPTION
**Pull request by jessiemongeon1**
Thank you for your contribution to the IC Developer Portal. This repo contains the content for https://internetcomputer.org and the ICP Developer Documentation, https://internetcomputer.org/docs/. 

If you are submitting a Pull Request for adding or changing content on the ICP Developer Documentation, please make sure that your contribution meets the following requirements:

- [x] Follows the [developer docs style guide](style-guide.md).
- [x] Follows the [best practices and guidelines](README.md#best-practices).
- [x] New documentation pages include [document tags](README.md#document-tags).
- [x] New documentation pages include [SEO keywords](README#seo-keywords).
- [x] New documents are in the `.mdx` file format to support the previous two components. 
- [x] New documents are registered in [`/sidebars.js`](https://github.com/dfinity/portal/blob/master/sidebars.js), otherwise, it will not appear in the
  side navigation bar.
- [x] Make sure that the [`.github/CODEOWNERS`](https://github.com/dfinity/portal/blob/master/.github/CODEOWNERS) file is
  filled with new documents that you added. This way we can ensure that future Pull Requests are reviewed by the right people.

### CODE REVIEW

**Review by dfx-json — APPROVED** [review]
Approving, but it would be good to get additional expert approval

File: docs/developer-docs/identity/authentication/msq.mdx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# MSQ
+
+<MarkdownChipRow labels={["Advanced", "Authentication"]} />
+
+## Overview
+
+[MSQ](https://github.com/fort-major/msq/tree/master) is a MetaMask Snap wallet designed authentication and integration with dapps deployed on ICP. MSQ uses a scoped identity architecture, meaning each user gets a unique and dedicated identity for each dapp or website they interact with. Scoped-identity architectures help protect users from signature-stealing attacks.
```
Comment (jennifertrin) [inline docs/developer-docs/identity/authentication/msq.mdx:13]:
```suggestion
[MSQ](https://github.com/fort-major/msq/tree/master) is a MetaMask Snap wallet designed authentication and integration with dapps deployed on ICP. MSQ uses a scoped-identity architecture, meaning each user gets a unique and dedicated identity for each dapp or website they interact with. Scoped-identity architectures help protect users from signature-stealing attacks.
```

File: docs/developer-docs/identity/authentication/msq.mdx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+```
+
+This function returns either a `bigint` or `null`. A `bigint` response will indicate the block number of the executed ICRC-1 transaction and means that the payment was successful. That block number can then be used to query the token canister, verify the payment, and execute some business logic.
+
+MSQ supports ICRC-1 token payments by default through a series of whitelisted ICRC-1 tokens. If you accept payment of a whitelisted token, your workflow will execute as expected. However, if you accept a token that isn't currently whitelisted, the user will be prompted to add the token to their personal whitelist.
+
+[View the MSQ documentation for the full list of whitelisted ICRC-1 tokens](https://github.com/fort-major/msq/blob/master/documentation/integration.md#icrc-1-payments).
+
+## Resources
+
```
Comment (jennifertrin) [inline docs/developer-docs/identity/authentication/msq.mdx:123]: I suggest that we also link MSQ in the Metamask Snap directory:https://snaps.metamask.io/snap/npm/fort-major/msq/.

File: docs/developer-docs/identity/authentication/msq.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jessiemongeon1) [reply docs/developer-docs/identity/authentication/msq.mdx]: Added link

File: docs/developer-docs/identity/authentication/msq.mdx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+It provides two primary features:
+
+- User authorization: Web services can sign arbitrary data with a user's scoped key pair. 
+
+- ICRC-1 payments: Users can request to perform ICRC-1 transfers to pay for goods or services.
+
+## Installation
+
+To use MSQ, import it into your project's `package.json` file:
```
[the remark this answers is not in the document]
Comment (jessiemongeon1) [reply docs/developer-docs/identity/authentication/msq.mdx:23]: Added install with npm line

File: docs/developer-docs/identity/authentication/msq.mdx (hunk 2 above)
Comment (jessiemongeon1) [reply docs/developer-docs/identity/authentication/msq.mdx:125]: Added link

File: docs/developer-docs/identity/authentication/msq.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jennifertrin) [reply docs/developer-docs/identity/authentication/msq.mdx]: It sounds like scoped-identity architecture means that identities are tied to a domain, not canister (similar to II). A MetaMask user connecting to the MSQ Snap would receive a different identity for each domain they interact with.

I'll let you decide what is the best way to explain this!

Response from MSQ team: https://forum.dfinity.org/t/msq-hits-the-metamask-snap-store/30499/23

**Review by jennifertrin — APPROVED** [review]

**Pull request merged**